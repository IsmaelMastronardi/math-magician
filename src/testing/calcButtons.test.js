import { render } from '@testing-library/react';
import CalculatorButtons from '../components/CalcButtons';

describe('Renders buttons for the calculator', () => {
  test('render', () => {
    const calcSymbols = [
      { symbol: 'AC', class: 'greyButton' },
      { symbol: '+/-', class: 'greyButton' },
      { symbol: '%', class: 'greyButton' },
      { symbol: 'รท', class: 'orangeButton' },
      { symbol: '7', class: 'greyButton' },
      { symbol: '8', class: 'greyButton' },
      { symbol: '9', class: 'greyButton' },
      { symbol: 'x', class: 'orangeButton' },
      { symbol: '4', class: 'greyButton' },
      { symbol: '5', class: 'greyButton' },
      { symbol: '6', class: 'greyButton' },
      { symbol: '-', class: 'orangeButton' },
      { symbol: '1', class: 'greyButton' },
      { symbol: '2', class: 'greyButton' },
      { symbol: '3', class: 'greyButton' },
      { symbol: '+', class: 'orangeButton' },
      { symbol: '0', class: 'greyButton0' },
      { symbol: '.', class: 'greyButton' },
      { symbol: '=', class: 'orangeButton' },
    ];
    const handleClick = () => {
    };
    const { buttons } = render(<CalculatorButtons arr={calcSymbols} handleClick={handleClick} />);
    expect(buttons).toMatchSnapshot();
  });
});
