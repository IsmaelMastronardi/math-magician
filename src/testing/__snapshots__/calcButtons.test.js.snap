// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders buttons for the calculator render 1`] = `undefined`;
